File: Programmers/Java/모음사전.java
import java.util.*;

class Solution {
    
    private static void go(String word,String vowels,int idx,List<String> l){
        if(idx > vowels.length()) return;
        l.add(word);
        
        for(int i = 0; i < vowels.length(); i++)
            go(word+vowels.charAt(i),vowels,idx+1,l);
    }
    
    public int solution(String word) {
        List<String> l = new ArrayList();
        String vowels = "AEIOU";
        
        go("",vowels,0,l);
   
        return l.indexOf(word);
    }
}

File: Programmers/Python/모음사전.py
def go(word,vowels,idx,l):
    if idx > len(vowels):
        return
    l.append(word)

    for i in range(0,len(vowels)):
        go(word+vowels[i],vowels,idx+1,l)

def solution(word):
    vowels = 'AEIOU'
    l = []
    go("",vowels,0,l)

    return l.index(word)
